lte/gateway/docker/mme/Dockerfile

```diff
@@ -0,0 +1,369 @@
+    echo "Install FreeDiameter requirements" && \
+    apt-get -y install libsctp1 libsctp-dev libgnutls28-dev libgcrypt-dev \
+    bison flex libidn11-dev && \
+    echo "Install libgtpnl requirements" && \
+    apt-get -y install libmnl-dev && \
+    echo "Install Nettle requirements" && \
+    apt-get install -y libgoogle-glog-dev libconfig-dev libxml2-dev \
+    libyaml-cpp-dev nlohmann-json-dev && \
+    echo "Install Prometheus requirements" && \
+    apt-get -y install telegraf && \
+    echo "Install ZeroMQ" && \
+    apt-get install -y libczmq-dev=4.1.0-2 && \
+    ln -s /usr/bin/python2.7 /usr/local/bin/python
+
+RUN ["/bin/bash", "-c", "if [[ -v GIT_PROXY ]]; then git config --global http.proxy $GIT_PROXY; fi"]
+
+##### GRPC and it's dependencies
+RUN git clone --recurse-submodules -b v1.15.0 https://github.com/grpc/grpc && \
+    echo "Install c-ares" && \
+    cd grpc && \
+    cd third_party/cares/cares && \
+    git fetch origin && \
+    git checkout cares-1_13_0 && \
+    mkdir -p cmake/build && \
+    cd cmake/build && \
+    cmake -DCMAKE_BUILD_TYPE=Release ../.. && \
+    make -j4 && \
+    make install && \
+    cd ../../../../.. && \
+    rm -rf third_party/cares/cares && \
+    echo "Install zlib" && \
+    cd third_party/zlib && \
+    mkdir -p cmake/build && \
+    cd cmake/build && \
+    cmake -DCMAKE_BUILD_TYPE=Release ../.. && \
+    make -j4 && \
+    make install && \
+    cd ../../../.. && \
+    rm -rf third_party/zlib && \
+    echo "Install protobuf" && \
+    cd third_party/protobuf && \
+    git submodule update --init --recursive  && \
+    ./autogen.sh  && \
+    ./configure  && \
+    make -j4 && \
+    make install && \
+    cd ../.. && \
+    rm -rf third_party/protobuf && \
+    ldconfig && \
+    echo "Install GRPC" && \
+    mkdir -p cmake/build && \
+    cd cmake/build && \
+    cmake \
+        -DgRPC_INSTALL=ON \
+        -DBUILD_SHARED_LIBS=ON \
+        -DgRPC_BUILD_TESTS=OFF \
+        -DgRPC_PROTOBUF_PROVIDER=package \
+        -DgRPC_ZLIB_PROVIDER=package \
+        -DgRPC_CARES_PROVIDER=package \
+        -DgRPC_SSL_PROVIDER=package \
+        -DCMAKE_BUILD_TYPE=Release \
+        ../.. && \
+    make -j4 && \
+    make install
+
+##### Prometheus CPP
+RUN git clone https://github.com/jupp0r/prometheus-cpp.git && \
+    cd prometheus-cpp && \
+    git checkout d8326b2bba945a435f299e7526c403d7a1f68c1f && \
+    git submodule init && git submodule update && \
+    mkdir _build && \
+    cd _build/ && \
+    cmake .. && \
+    make -j4 && \
+    make install
+
+##### Redis CPP
+RUN git clone https://github.com/cpp-redis/cpp_redis.git && \
+    cd cpp_redis && \
+    git checkout bbe38a7f83de943ffcc90271092d689ae02b3489 && \
+    git submodule init && git submodule update && \
+    mkdir build && cd build && \
+    cmake .. -DCMAKE_BUILD_TYPE=Release && \
+    make -j4 && \
+    make install
+
+##### NETTLE
+RUN wget https://ftp.gnu.org/gnu/nettle/nettle-2.5.tar.gz && \
```

Comment:
@119Vik 

I've never asked but why are we install nettle from source? Is it required to have 2.5?

Raphael

Reply:
@rdefosse When I was looking for dependencies and their versions I used vagrant AGW box defined in the repo as a source of truth.
In the box we have :
```
$ dpkg -l | grep nett
ii  libnettle6:amd64                     3.3-1+b2                          amd64        low level cryptographic library (symmetric and one-way cryptos)
ii  oai-nettle                           2.5-1                             amd64        A low-level cryptographic library
```
oai-nettle 2.5 , as well as libnettle 2.5, is not in the default ubuntu repo so I built it from sources. In case oai-nettle is orphan package and the only thing we need is the latest libnettle - I'm happy to replace this "build from sources stage" with one more item in the "apt-get install" stage.

Reply:
@119Vik 

OK. Let's keep it this way. We will make optimization later.